src/Database/Bloodhound/Types.hs

```diff
@@ -5212,3 +5213,10 @@ instance FromJSON NodeNetworkInterface where
       parse o = NodeNetworkInterface <$> o .: "mac_address"
                                      <*> o .: "name"
                                      <*> o .: "address"
+
+newtype MaybeNA a = MaybeNA { unMaybeNA :: Maybe a }
+  deriving (Show, Eq)
+
+instance FromJSON a => FromJSON (MaybeNA a) where
+  parseJSON (String "NA") = pure $ MaybeNA Nothing
+  parseJSON o = MaybeNA . Just <$> parseJSON o
```

Comment:
I recommended it be done this way. To me, NA is semantically meant to indicate that the version is not available. It would make sense to me if I saw it in the haddocks, plus I get to use all of my tools such as `maybe` and the `Maybe` monad to work with the value still whereas all I've got is a `case` if we added a constructor for NA version.